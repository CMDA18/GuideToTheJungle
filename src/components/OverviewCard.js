// @flow
import React, { Fragment } from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { textStyles, colorPalette } from '../styles/styleGuide'
import LabelBestPractice from '../shared/elements/LabelBestPractice'

type Props = {
  slug: string,
  image: string,
  order: number,
  title: string,
  text: string,
  unlockText?: string,
  unlocked: 'yes' | 'no'
}

const OverviewCard = (props: Props) => (
  <Fragment>
    <Card to={props.slug} unlocked={props.unlocked}>
      {props.image &&
        <Wrapper>
          <ImageContainer unlocked={props.unlocked}>
            <Image
              backgroundImage={
                props.image ? props.image : props.image
              }
            />
          </ImageContainer>
          <LabelContainer unlocked={props.unlocked}>
            <LabelBestPractice unlockText={props.unlockText}
            />
          </LabelContainer>
        </Wrapper>
      }
      <TextCard unlocked={props.unlocked}>
        <Module>Module {props.order}</Module>
        <Title>{props.title}</Title>
        <Text>{props.text}</Text>
      </TextCard>
    </Card>

  </Fragment>
)

export default OverviewCard

const Module = styled.h2`
  margin-top: 8px;
  margin-bottom: 0;
  transition: color 0.3s ease-in-out;
  ${textStyles.title};
`

const Title = styled.h3`
  margin-top: 0;
  margin-bottom: 20px;
  transition: color 0.3s ease-in-out;
  ${textStyles.title};
`

const Image = styled.div`
  transform: scale(1);
  transform-origin: center center;
  transition: transform 0.3s ease-in-out;
  will-change: transform;
  backface-visibility: hidden;
  background-position: center center;
  background-image: url(${props =>
    props.backgroundImage ? props.backgroundImage : ''});
  background-size: cover;
  height: 206px;
`

const Wrapper = styled.div`
  transform: translate3d(0, 0, 0);
  overflow: hidden;
`

const Card = styled(Link)`
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.15);
  border-radius: 4px;
  background-color: #ffffff;
  overflow: hidden;
  text-decoration: none;
  pointer-events: ${props => (props.unlocked === 'yes' ? `auto` : `none`)};

  &:hover {
    ${Module} {
      color: ${colorPalette.orange};
    }
    ${Title} {
      color: ${colorPalette.orange};
    }
    ${Image} {
      transform: scale(1.08);
    }
  }
`

const ImageContainer = styled.div`
  transform: translate3d(0, 0, 0);
  border-radius: 4px 4px 0 0;
  height: 206px;
  overflow: hidden;
  opacity: ${props => (props.unlocked === 'yes' ? `1` : `0.5`)};
`

const TextCard = styled.div`
  min-height: 150px;
  background-color: #ffffff;
  padding: 0 16px 0px 16px;
  opacity: ${props => (props.unlocked === 'yes' ? `1` : `0.5`)};
`

const Text = styled.p``

const LabelContainer = styled.div`
  right: -43px;
  top: 55px;
  position: absolute;
  transform: rotate(37deg);
  display: ${props => (props.unlocked === 'yes' ? `none` : `block`)};
`
