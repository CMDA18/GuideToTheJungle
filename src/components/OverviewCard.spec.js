import React from 'react'
import { shallow } from 'enzyme'
import OverviewCard from './OverviewCard'

it('renders without crashing', () => {
  expect(shallow(<OverviewCard />)).toMatchSnapshot()
})
