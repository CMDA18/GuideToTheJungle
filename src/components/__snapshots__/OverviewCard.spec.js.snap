// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OverviewCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <OverviewCard__Card
        to={undefined}
        unlocked={undefined}
      >
        <OverviewCard__TextCard
          unlocked={undefined}
        >
          <OverviewCard__Title />
          <OverviewCard__Text />
        </OverviewCard__TextCard>
      </OverviewCard__Card>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          undefined,
          <OverviewCard__TextCard
            unlocked={undefined}
          >
            <OverviewCard__Title />
            <OverviewCard__Text />
          </OverviewCard__TextCard>,
        ],
        "to": undefined,
        "unlocked": undefined,
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <OverviewCard__Title />,
              <OverviewCard__Text />,
            ],
            "unlocked": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <OverviewCard__Card
          to={undefined}
          unlocked={undefined}
        >
          <OverviewCard__TextCard
            unlocked={undefined}
          >
            <OverviewCard__Title />
            <OverviewCard__Text />
          </OverviewCard__TextCard>
        </OverviewCard__Card>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            undefined,
            <OverviewCard__TextCard
              unlocked={undefined}
            >
              <OverviewCard__Title />
              <OverviewCard__Text />
            </OverviewCard__TextCard>,
          ],
          "to": undefined,
          "unlocked": undefined,
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <OverviewCard__Title />,
                <OverviewCard__Text />,
              ],
              "unlocked": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
